src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,46 @@
+package core.basesyntax.service;
+
+import core.basesyntax.FruitTransaction;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        Map<String, FruitTransaction> calculatedTransactionsMap = new HashMap<>();

```

Comment:
You need to store the current balance of fruits in a separate class Storage that will represent the database